knotx-core/src/main/java/io/knotx/launcher/KnotxStarterVerticle.java

```diff
@@ -33,19 +33,27 @@
   @Override
   public void start(Future<Void> startFuture) throws Exception {
     printLogo();
-    Observable.from(config().getJsonArray("modules"))
-        .flatMap(this::deployVerticle)
-        .compose(joinDeployments())
-        .subscribe(
-            message -> {
-              LOGGER.info("Knot.x STARTED {}", message);
-              startFuture.complete();
-            },
-            error -> {
-              LOGGER.error("Verticle could not be deployed {}", error);
-              startFuture.fail(error);
-            }
-        );
+    Observable.defer(() -> Observable.from(config().getJsonArray("modules"))
```

Comment:
I'm not sure `defer` is necessary here. We have only single subscriber in this situation, and created `Observable` is not exposed outside this method.